modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/autocomplete/Autocomplete.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayAutocomplete from '@clayui/autocomplete';
+import ClayDropDown from '@clayui/drop-down';
+import {FocusScope} from '@clayui/shared';
+import React, {useEffect, useRef, useState} from 'react';
+
+const DefaultItemsWrapperRenderer = ({
+	items,
+	labelKey,
+	onItemClick,
+	valueKey,
+}) => {
```

Comment:
Should we provide some defaults here, at least for maybe `labelKey` and `valueKey`?

I see some disconnect between this and how it is eventually used in this case

```
<BaseAutocomplete
	labelKey={itemsLabel}
	valueKey={itemsKey}
/>
```

Which in turn goes up the chain. I think it might be too easy to forget to pass things that are required for the default behaviour to work properly, making the default a bit less useful.

A slight improvement would be to have this APIs typed. Then, at the very least, one could know what's needed ahead of time. #justsayin ;)

Reply:
I was trying to change as little as possible on the commerce side which is why there is a disconnect. But now I am just thinking I should re-write the commerce component to be a bit more in-line with our existing patterns.